command/agent/job_endpoint.go

```diff
@@ -1258,11 +1258,16 @@ func ApiServicesToStructs(in []*api.Service) []*structs.Service {
 					Body:                   check.Body,
 					GRPCService:            check.GRPCService,
 					GRPCUseTLS:             check.GRPCUseTLS,
-					TaskName:               check.TaskName,
 					SuccessBeforePassing:   check.SuccessBeforePassing,
 					FailuresBeforeCritical: check.FailuresBeforeCritical,
 					OnUpdate:               onUpdate,
 				}
+
+				if group {
+					// only copy over task name for group level checks
+					out[i].Checks[j].TaskName = check.TaskName
+				}
```

Comment:
🤦 Missing this in review of that refactoring was almost inevitable for me. Good catch on this fix.